server.js

```diff
@@ -424,6 +424,20 @@ app.post('/api/v1/users', jsonParser, function(req, res) {
   });
 });
 
+app.post('/api/v1/users/:username', jsonParser, function(req, res) {
+  auth.check(req, 'can_update_users', null, function(err) {
+    if (err) {
+      return serverUtils.error(err, req, res);
+    }
+    users.updateUser(req.params.username, req.body, function(err) {
+      if (err) {
+        return serverUtils.serverError(err, req, res);
```

Comment:
I think serverError is a bit aggressive - usually this is reserved for unexpected exceptions (like bugs in the code, out of memory, etc). For example, we don't need to output to console.error just because someone supplied invalid args. Try and work out what the right http status code is and return a useful JSON response with `success: false` or something. Perhaps where the error is generated you could return a JS object instead, for example instead of...

```
return callback(new Error(
  'One of the following fields are required: type, password or place.'
));
```

You could use...

```
return callback({
  code: 400,
  message: 'One of the following fields are required: type, password or place.'
});
```

There is code in `serverUtils.error` to handle this but it would need updating if you choose to use it.